File: src/MiniErp.Web/src/types/venda.ts
export interface ItemVenda {
  produtoId: number
  produtoNome: string
  quantidade: number
  precoUnitario: number
  subtotal: number
}

export interface Venda {
  id: number
  clienteId: number
  clienteNome: string
  data: string
  total: number
  itens: ItemVenda[]
}

export interface CriarItemVendaRequest {
  produtoId: number
  quantidade: number
}

export interface CriarVendaRequest {
  clienteId: number
  itens: CriarItemVendaRequest[]
}

File: src/MiniErp.Domain/Entities/Venda.cs
namespace MiniErp.Domain.Entities;

public class Venda
{
    public int Id { get; set; }

    public int ClienteId { get; set; }
    
    public Cliente Cliente { get; set; } = null!;

    public DateTime Data { get; set; }

    public List<ItemVenda> Itens { get; set; } = [];

    public decimal Total
    {
        get
        {
            return Itens.Sum(item => item.Subtotal);
        }
    }
}